frontend/app/utils/route-utils.ts

```diff
@@ -1,7 +1,31 @@
+import type { Params } from 'react-router';
+import { generatePath, redirect } from 'react-router';
+
 import { AppError } from '~/errors/app-error';
 import { ErrorCodes } from '~/errors/error-codes';
 import type { I18nPageRoute, I18nRoute, I18nRouteFile } from '~/i18n-routes';
-import { isI18nLayoutRoute, isI18nPageRoute } from '~/i18n-routes';
+import { i18nRoutes, isI18nLayoutRoute, isI18nPageRoute } from '~/i18n-routes';
+import { getLanguage } from '~/utils/i18n-utils';
+
+/**
+ * Generate a redirect response. Sets the status code and the Location header. Defaults to "302 Found".
+ *
+ * @param i18nRouteFile - The file name of the route to redirect to.
+ * @param resource - The request, URL, or path to extract the language from.
+ * @param params - Values for any parameters that are required in the URL.
+ */
+export function appRedirect(i18nRouteFile: I18nRouteFile, resource: Request | URL | string, params?: Params): Response {
+  const language = getLanguage(resource);
+
+  if (language === undefined) {
+    throw new AppError('No language found in request', ErrorCodes.NO_LANGUAGE_FOUND);
+  }
```

Comment:
You can't use it to redirect to a non-bilingual route?

Reply:
Non-bilingual routes don't have the special i18n routing stuff.. for those you can just use `redirect()`

Reply:
Maybe this fn should be called `i18nRedirect()`?

Reply:
I'm not sure. What make more sense to you. I'm trusting your judgment, as usual.

Reply:
[Renamed to `i18nRedirect()`](https://github.com/DTS-STN/future-sir/pull/122/files#diff-d82ba9d8cbbfa33f03394e372c60135e475d907ae165f136e2c55544e6c3a8ffR18)